### ISSUE TITLE
AutoLoader Customizing Inflections in initializers dir is not working for models in autoloaded_once_paths

### ISSUE THREAD

**Issue by cantin** [rails/rails#45568]
[The customizing inflections section in Autoloading Rails guide](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#customizing-inflections) is not working for classes in autoloaded_once_paths.

Because the initializer `setup_once_autoloader` is run before the `load_config_initializers`.

### Steps to reproduce

Run following script:
```bash
rails new testapp && cd testapp
# Add autoload_paths and autoload_once_paths in config/application.rb
ruby -i -pe 'gsub(/(class Application.+)/) { $1 + "\n config.autoload_once_paths += %W(autoload_once) \n config.autoload_paths += %W(autoload)" }' config/application.rb

mkdir autoload_once autoload
echo "class AI; end" > autoload_once/ai.rb
echo "class CI; end" > autoload/ci.rb

touch config/initializers/zeitwerk.rb

# add inflector
cat <<-EOF > config/initializers/zeitwerk.rb
  Rails.autoloaders.each do |autoloader|
    autoloader.inflector.inflect(
      "ai"   => "AI",
      "ci"   => "CI"
    )
  end
EOF

# output CI successfully
bundle exec rails r 'puts CI'

# Error occurred: uninitialized constant AI
bundle exec rails r 'puts AI'
```

### Expected behavior
`bundle exec rails r 'puts AI'` should output AI

### Actual behavior
Error occurred:  uninitialized constant AI

### System configuration
**Rails version**: 7.0.3

**Ruby version**: 3.1.2

**Comment by skipkayhil** [comment]
Hey @cantin, thanks for the report! I agree that this behavior is confusing, however I'm not sure if its possible to change. The once autoloader is specifically documented as being used for autoloading things (once) in initializers, so it doesn't seem like it should be possible to modify the inflector in an initializer (as you discovered). Maybe the docs should be updated to recommend configuring the inflector in `config/application.rb` instead of an initializer?

**Comment by cantin** [comment]
Hey @skipkayhil, thanks for the reply. Updating the docs sounds great to me.

**Comment by skipkayhil** [comment]
> Updating the docs sounds great to me.

Awesome, would you be interested in creating a PR?

**Comment by skipkayhil** [comment]
Closed by #45581
